done third task

BarsTK opened the pull request.

NikitaSedyx commented on task3/task3.html:
```diff
@@ -0,0 +1,56 @@
+<!-- perform binary search in sorted number array-->
+  <meta charset="utf-8">
+  <script type="text/javascript">
+    function binSearch(a, b) {
```
just curious, why do you put your script in html?

NikitaSedyx commented on task3/task3.html:
```diff
@@ -0,0 +1,56 @@
+  <meta charset="utf-8">
+  <script type="text/javascript">
+    function binSearch(a, b) {
+      var x = a.split(',');
```
hm, a should be array

BarsTK commented:
I took note of the comments and removed main function to separate file.
About html: to you could check function without console. Do I need another
way?

On Sun, Apr 9, 2017 at 6:48 PM, Mikita Siadykh <notifications@github.com>
wrote:

> *@NikitaSedyx* requested changes on this pull request.
>
> search doesn't work when only 1 item in array + it's time to start naming
> variables with sense
> ------------------------------
>
> In task3/task3.html
> <https://github.com/BarsTK/BarsTK.github.io/pull/3#discussion_r110545255>:
>
> > @@ -0,0 +1,56 @@
> +<!DOCTYPE html>
> +<html>
> +<head>
> +<!-- perform binary search in sorted number array-->
> +  <meta charset="utf-8">
> +  <script type="text/javascript">
> +    function binSearch(a, b) {
>
> just curious, why do you put your script in html?
> ------------------------------
>
> In task3/task3.html
> <https://github.com/BarsTK/BarsTK.github.io/pull/3#discussion_r110545285>:
>
> > @@ -0,0 +1,56 @@
> +<!DOCTYPE html>
> +<html>
> +<head>
> +<!-- perform binary search in sorted number array-->
> +  <meta charset="utf-8">
> +  <script type="text/javascript">
> +    function binSearch(a, b) {
> +      var x = a.split(',');
>
> hm, a should be array
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BarsTK/BarsTK.github.io/pull/3#pullrequestreview-31727206>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AZGgMxxRCzYn6NWAEHSMpBPhX5Ny_uojks5ruP3OgaJpZM4M2Paq>
> .
>

NikitaSedyx commented:
@BarsTK 
your binSearch should take sorted array as first parameter
in execBin.. you can transform your string to array... it's simple

NikitaSedyx commented on task3/task3.html:
```diff
@@ -0,0 +1,26 @@
+      var val = document.getElementById("number").value;
+      const result = binSearch(arr, +val);
+      
+      alert(result == -1 ? "Element is not found" :"Searched number is in position with index = " + result);
```
nit: ` : ` spaces

NikitaSedyx commented on task3/search.js:
```diff
@@ -0,0 +1,30 @@
+function binSearch(arr, searchNumb) {
+  var len = arr.length;
+  const ascFlag = (arr[0] < arr[len - 1]);
```
there is practise to name constants NAME_OF_CONSTANT upper snake case

NikitaSedyx commented on task3/search.js:
```diff
@@ -0,0 +1,30 @@
+  const diffIdxPos = 1;
+  var leftIdx = 0;
+  var rightIdx = len - 1;
+  var searchedIdx = -1;
```
if you use const you can use let

NikitaSedyx commented on task3/search.js:
```diff
@@ -0,0 +1,35 @@
+  const DIFF_IDX_POS = 1;
+  var len = arr.length;
+  var leftIdx = 0;
+  var rightIdx = len - 1;
```
I mean if you use const you can use let instead of var. and not let instead of const :)
